iothub/device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/mqtt/Mqtt.java

```diff
@@ -327,8 +327,6 @@ public IotHubTransportMessage receive()
     @Override
     public void connectionLost(Throwable throwable)
     {
-        log.warn("Mqtt connection lost", throwable);
```

Comment:
Removing this, you will not get noticed about disconnecting. In addition, log it as debug and in case where listener is not null, maybe suppose an information lost.

Reply:
```Listener``` is never null here, so moving this line lower doesn't risk losing any log statements

Reply:
Hmm, so we can remove `if (this.listener != null)` and leave as `log.info`? Or even better do this logic:
```
if (transportException.isRetryable()) 
{
    log.info("Mqtt connection lost");
}
else
{
    log.warn("Mqtt connection lost", throwable);
}
```
to avoid add exception when log as info.

Reply:
MQTT level logs generally aren't done at an Info level because the SDK abstracts most of that logic. The reconnection logic layer will do any info level logging related to disconnects.

> Or even better do this logic

I'm not sure I follow why we would care about retryable vs non-retryable when logging that the connection was lost. I can remove the ```if (this.listener != null)``` though, sure.

Reply:
From our point of view, a retryable exception is not the final error, so maybe, when retrying, you can log it as info otherwise log as warn or debug.

If you don't want to add the "if logic" in the `log.debug`, instead of adding the full exception object as a parameter you can append the message of the exception in order to avoid posting an exception in telemetry.

Reply:
> instead of adding the full exception object as a parameter you can append the message of the exception in order to avoid posting an exception in telemetry.

I'd rather not. Omitting the full stacktrace here would make it harder to root-cause disconnects on the off-chance that the exception doesn't make it up to the reconnection logic where it would normally get logged.

The fact that this MQTT-layer log with a stacktrace is done at a Debug level means you can still "hide" this log (and its full stacktrace) by only listening for Info level logs.

Reply:
Ok, close this review.